PracticeRobot2/src/main/java/frc/robot/Robot.java

```diff
@@ -0,0 +1,203 @@
+
+      // Connect the simulators with their counterparts
+      m_leftEncoderSim = new EncoderSim(m_leftEncoder);
+      m_rightEncoderSim = new EncoderSim(m_rightEncoder);
+      m_gyroSim = new AnalogGyroSim(m_gyro);
+    }
+  }
+
+  /**
+   * This function is called every robot packet, no matter the mode. Use this for
+   * items like diagnostics that you want ran during disabled, autonomous,
+   * teleoperated and test.
+   *
+   * <p>
+   * This runs after the mode specific periodic functions, but before LiveWindow
+   * and SmartDashboard integrated updating.
+   */
+  @Override
+  public void robotPeriodic() {
+  }
+
+  @Override
+  public void simulationPeriodic() {
+    // connect the motors to update the drivetrain
+    m_driveTrainSim.setInputs(m_leftMotor.get() * RobotController.getBatteryVoltage(),
+        m_rightMotor.get() * RobotController.getBatteryVoltage());
+
+    // Run and update simulation
+    m_driveTrainSim.update(0.02);
+    m_leftEncoderSim.setDistance(m_driveTrainSim.getLeftPositionMeters());
+    m_leftEncoderSim.setRate(m_driveTrainSim.getLeftVelocityMetersPerSecond());
+    m_rightEncoderSim.setDistance(m_driveTrainSim.getRightPositionMeters());
+    m_rightEncoderSim.setRate(m_driveTrainSim.getRightVelocityMetersPerSecond());
+    m_gyroSim.setAngle(-m_driveTrainSim.getHeading().getDegrees());
+
+    m_odometry.update(Rotation2d.fromDegrees(getHeading()), m_leftEncoder.getDistance(), m_rightEncoder.getDistance());
+    m_fieldSim.setRobotPose(getPose());
+  }
+
+  /**
+   * This autonomous (along with the chooser code above) shows how to select
+   * between different autonomous modes using the dashboard. The sendable chooser
+   * code works with the Java SmartDashboard. If you prefer the LabVIEW Dashboard,
+   * remove all of the chooser code and uncomment the getString line to get the
+   * auto name from the text box below the Gyro
+   *
+   * <p>
+   * You can add additional auto modes by adding additional comparisons to the
+   * switch structure below with additional strings. If using the SendableChooser
+   * make sure to add them to the chooser code above as well.
+   */
+  @Override
+  public void autonomousInit() {
+    m_autoSelected = m_chooser.getSelected();
+    // m_autoSelected = SmartDashboard.getString("Auto Selector", kDefaultAuto);
+    System.out.println("Auto selected: " + m_autoSelected);
+  }
+
+  /** This function is called periodically during autonomous. */
+  @Override
+  public void autonomousPeriodic() {
+    switch (m_autoSelected) {
+      case kCustomAuto:
+        // Put custom auto code here
+        break;
+      case kDefaultAuto:
+      default:
+        // Put default auto code here
+        break;
+    }
+  }
+
+  /** This function is called once when teleop is enabled. */
+  @Override
+  public void teleopInit() {
+  }
+
+  /** This function is called periodically during operator control. */
+  @Override
+  public void teleopPeriodic() {
+  }
```

Comment:
So right now, the robot won't do anything. That's because the joystick doesn't have a way to communicate with the DifferentialDrive (where all the motors are). 

All you have to do is add the line:
m_robotDrive.arcadeDrive(m_stick.getX(), m_stick.getY());
inside teleopPeriodic.

That way the robot will be controlled using joysticks.